sections/dir.zsh

```diff
@@ -31,6 +35,10 @@ spaceship_dir() {
     dir="%${SPACESHIP_DIR_TRUNC}~"
   fi
 
+  if [[ ! -w . ]]; then
+    SPACESHIP_DIR_SUFFIX="%F{$SPACESHIP_DIR_LOCK_COLOR} $SPACESHIP_DIR_LOCK_SYMBOL %f"
```

Comment:
Since you are using this approach, I suggest these improvements:

1. Remove space here before `$SPACESHIP_DIR_LOCK_SYMBOL`, instead add space before the lock icon in the default value of `SPACESHIP_DIR_LOCK_SYMBOL` above - this will allow this:

    ![image](https://user-images.githubusercontent.com/1177900/35197108-5df703c0-fedb-11e7-86b8-217936ac1e2a.png)

2. You are overriding the value of `$SPACESHIP_DIR_SUFFIX`, I think you should keep it in case users configured it.

So this line I guess should become something like:

```
SPACESHIP_DIR_SUFFIX="%F{$SPACESHIP_DIR_LOCK_COLOR}$SPACESHIP_DIR_LOCK_SYMBOL%f$SPACESHIP_DIR_SUFFIX"
```

Reply:
@phineas0fog I updated the last line ^^ a little after publishing, just so you know.

For readability, I think this is the best:

```
SPACESHIP_DIR_SUFFIX="%F{$SPACESHIP_DIR_LOCK_COLOR}${SPACESHIP_DIR_LOCK_SYMBOL}%f${SPACESHIP_DIR_SUFFIX}"
```